shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/controllers/BridgeExecutorController.java

```diff
@@ -76,12 +78,32 @@ public List<EventSource> prepareEventSources(EventSourceContext<BridgeExecutor>
 
     @Override
     public UpdateControl<BridgeExecutor> reconcile(BridgeExecutor bridgeExecutor, Context context) {
-        LOGGER.debug("Create or update BridgeProcessor: '{}' in namespace '{}'", bridgeExecutor.getMetadata().getName(), bridgeExecutor.getMetadata().getNamespace());
+        LOGGER.info("Create or update BridgeProcessor: '{}' in namespace '{}'", bridgeExecutor.getMetadata().getName(), bridgeExecutor.getMetadata().getNamespace());
+
+        BridgeExecutorStatus status = bridgeExecutor.getStatus();
+        status.getConditions().forEach(c -> LOGGER.debug("BridgeExecutor '{}' in namespace '{}' condition: {}:{}:{}",
+                bridgeExecutor.getMetadata().getName(),
+                bridgeExecutor.getMetadata().getNamespace(),
+                c.getType(),
+                c.getStatus(),
+                c.getReason()));
+
+        // Mark CRD as being reconciled
+        if (status.getConditionByType(ConditionTypeConstants.READY)
+                .filter(c -> c.getStatus() != ConditionStatus.False)
+                .isPresent()) {
+            LOGGER.info("Marking bridgeExecutor '{}' in namespace '{}' as augmenting reconciliation.",
+                    bridgeExecutor.getMetadata().getName(),
+                    bridgeExecutor.getMetadata().getNamespace());
+            bridgeExecutor.getStatus().markConditionFalse(ConditionTypeConstants.READY);
+            bridgeExecutor.getStatus().markConditionTrue(ConditionTypeConstants.AUGMENTATION, ConditionReasonConstants.RECONCILIATION_PROGRESSING);
+            return UpdateControl.updateStatus(bridgeExecutor);
+        }
```

Comment:
Why do we need this?

Reply:
See my comment above.

I won't reply to the other minor comments as I am more interested in your thoughts of the the high-level approach first.